### PR TITLE
Add conan source command to client

### PR DESCRIPTION
**Pull request by tru**
This runs the source method in the conanfile. It also accepts the -f flag
to force removal and re-running the source method.

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -366,6 +366,23 @@ def package(self, *args):
...
+        parser = argparse.ArgumentParser(description=self.source.__doc__, prog="conan source")
+        parser.add_argument("reference", help="package recipe reference name. e.g., zlib-ng/1.2.8@plex/stable")
+        parser.add_argument("-f", "--force", default=False, action="store_true", help="force remove the source directory and run again.")
+
+        args = parser.parse_args(*args)
+
+        try:
+            reference = ConanFileReference.loads(args.reference)
+        except:
+            raise ConanException("Invalid package recipe refenence. e.g., zlib-ng/1.2.8@plex/stable")
```
Comment (raulbocanegra) [inline conans/client/command.py]: Seems that there is a typo.

Comment (tru) [inline conans/client/command.py]: Thanks, fixed.

**Comment by memsharded** [comment]
Looks good, I have added a test and proposed an internal refactor in https://github.com/tru/conan/pull/4

**Comment by tru** [comment]
Merged your change and ready to go!

**Pull request merged**